pkg/vz/vm_darwin.go

```diff
@@ -0,0 +1,485 @@
+	if err != nil {
+		return err
+	}
+	ciData, err := vz.NewVirtioBlockDeviceConfiguration(ciDataAttachment)
+	if err != nil {
+		return err
+	}
+	configurations = append(configurations, ciData)
+
+	diffDiskAttachment, err := vz.NewDiskImageStorageDeviceAttachment(diffDiskPath, false)
+	if err != nil {
+		return err
+	}
+
+	diffDisk, err := vz.NewVirtioBlockDeviceConfiguration(diffDiskAttachment)
+	if err != nil {
+		return err
+	}
+	configurations = append(configurations, diffDisk)
+
+	vmConfig.SetStorageDevicesVirtualMachineConfiguration(configurations)
+	return nil
+}
+
+func attachDisplay(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	if driver.Yaml.Video.Display != nil {
+		graphicsDeviceConfiguration, err := vz.NewVirtioGraphicsDeviceConfiguration()
+		if err != nil {
+			return err
+		}
+		scanoutConfiguration, err := vz.NewVirtioGraphicsScanoutConfiguration(1920, 1200)
+		if err != nil {
+			return err
+		}
+		graphicsDeviceConfiguration.SetScanouts(scanoutConfiguration)
+
+		vmConfig.SetGraphicsDevicesVirtualMachineConfiguration([]vz.GraphicsDeviceConfiguration{
+			graphicsDeviceConfiguration,
+		})
+	}
+	return nil
+}
+
+func attachConsole(_ *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	consoleDevice, err := vz.NewVirtioConsoleDeviceConfiguration()
+	if err != nil {
+		return err
+	}
+	spiceAgentAttachment, err := vz.NewSpiceAgentPortAttachment()
+	if err != nil {
+		return err
+	}
+	spiceAgentName, err := vz.SpiceAgentPortAttachmentName()
+	if err != nil {
+		return err
+	}
+	spiceAgentPort, err := vz.NewVirtioConsolePortConfiguration(
+		vz.WithVirtioConsolePortConfigurationAttachment(spiceAgentAttachment),
+		vz.WithVirtioConsolePortConfigurationName(spiceAgentName),
+	)
+	if err != nil {
+		return err
+	}
+
+	consoleDevice.SetVirtioConsolePortConfiguration(0, spiceAgentPort)
+	vmConfig.SetConsoleDevicesVirtualMachineConfiguration([]vz.ConsoleDeviceConfiguration{
+		consoleDevice,
+	})
+	return nil
+}
+
+func attachFolderMounts(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	mounts := make([]vz.DirectorySharingDeviceConfiguration, len(driver.Yaml.Mounts))
+	for i, mount := range driver.Yaml.Mounts {
+		expandedPath, err := localpathutil.Expand(mount.Location)
+		if err != nil {
+			return err
+		}
+		directory, err := vz.NewSharedDirectory(expandedPath, !*mount.Writable)
+		if err != nil {
+			return err
+		}
+		share, err := vz.NewSingleDirectoryShare(directory)
+		if err != nil {
+			return err
+		}
+
+		tag := fmt.Sprintf("mount%d", i)
+		config, err := vz.NewVirtioFileSystemDeviceConfiguration(tag)
+		if err != nil {
+			return err
+		}
+		config.SetDirectoryShare(share)
+		mounts[i] = config
+	}
+	vmConfig.SetDirectorySharingDevicesVirtualMachineConfiguration(mounts)
+	return nil
+}
+
+func attachOtherDevices(_ *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
```

Comment:
does it make sense to register there also `NewLinuxRosettaDirectoryShare` 

So we have Rosetta inside VZ? https://github.com/Code-Hex/vz/blob/c436fc367107393c6eda419ff0f69903f674c118/shared_directory_arm64.go#L63

Reply:
Can be another PR